[2025-07-08T16:45:38.664+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-07-08T16:45:38.704+0000] {taskinstance.py:2631} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: process_exercises.process_exercises manual__2025-07-08T16:45:33.858254+00:00 [queued]>
[2025-07-08T16:45:38.715+0000] {taskinstance.py:2631} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: process_exercises.process_exercises manual__2025-07-08T16:45:33.858254+00:00 [queued]>
[2025-07-08T16:45:38.716+0000] {taskinstance.py:2884} INFO - Starting attempt 1 of 1
[2025-07-08T16:45:38.759+0000] {taskinstance.py:2907} INFO - Executing <Task(SparkSubmitOperator): process_exercises> on 2025-07-08 16:45:33.858254+00:00
[2025-07-08T16:45:38.767+0000] {logging_mixin.py:190} WARNING - /home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py:70 DeprecationWarning: This process (pid=7249) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-07-08T16:45:38.769+0000] {standard_task_runner.py:72} INFO - Started process 7250 to run task
[2025-07-08T16:45:38.767+0000] {standard_task_runner.py:104} INFO - Running: ['airflow', 'tasks', 'run', 'process_exercises', 'process_exercises', 'manual__2025-07-08T16:45:33.858254+00:00', '--job-id', '163', '--raw', '--subdir', 'DAGS_FOLDER/process_exercises.py', '--cfg-path', '/tmp/tmplpirdpe6']
[2025-07-08T16:45:38.770+0000] {standard_task_runner.py:105} INFO - Job 163: Subtask process_exercises
[2025-07-08T16:45:38.886+0000] {task_command.py:467} INFO - Running <TaskInstance: process_exercises.process_exercises manual__2025-07-08T16:45:33.858254+00:00 [running]> on host 3748f1b8240a
[2025-07-08T16:45:39.176+0000] {taskinstance.py:3157} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='airflow' AIRFLOW_CTX_DAG_ID='process_exercises' AIRFLOW_CTX_TASK_ID='process_exercises' AIRFLOW_CTX_EXECUTION_DATE='2025-07-08T16:45:33.858254+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-07-08T16:45:33.858254+00:00'
[2025-07-08T16:45:39.177+0000] {taskinstance.py:740} INFO - ::endgroup::
[2025-07-08T16:45:39.217+0000] {base.py:84} INFO - Retrieving connection 'spark-ebsim'
[2025-07-08T16:45:39.225+0000] {spark_submit.py:474} INFO - Spark-Submit cmd: spark-submit --master spark://spark-master:7077 --jars /opt/bitnami/spark/jars/iceberg-spark-runtime-3.5_2.12-1.8.1.jar,/opt/bitnami/spark/jars/nessie-spark-extensions-3.5_2.12-0.103.2.jar,/opt/bitnami/spark/jars/nessie-client-0.103.2.jar,/opt/bitnami/spark/jars/hadoop-aws-3.3.2.jar,/opt/bitnami/spark/jars/aws-java-sdk-bundle-1.12.262.jar, --name arrow-spark --deploy-mode client /opt/bitnami/spark/jobs/process_exercises_job.py {"id": "CA-SUL-virtual-99RC-2025-07-08", "manifest_entry": {"data": {"createdAt": "2025-07-08T16:43:26.899Z", "origin": "CA-SUL", "path": "exercises/CA-SUL-virtual-99RC-2025-07-08"}, "id": "CA-SUL-virtual-99RC-2025-07-08"}, "metadata": {"createdAt": "2025-07-08T16:43:26.899Z", "description": "99RC", "documents": [{"description": "", "id": "1751993019656", "mimetype": "application/pdf", "name": "Op Jacui IV - Ordem de Operacoes.pdf", "path": "exercises/CA-SUL-virtual-99RC-2025-07-08/documents/Op Jacui IV - Ordem de Operacoes.pdf", "size": 336744, "uploadedAt": "2025-07-08T16:43:39.656Z"}], "endDate": "2025-07-10", "id": "CA-SUL-virtual-99RC-2025-07-08", "name": "Exerc\u00edcio Virtual", "origin": "CA-SUL", "simulationType": "virtual", "simulations": [{"id": "172.19.0.14_2025-07-08", "movedAt": "2025-07-08T16:43:43.163Z", "name": "172.19.0.14_2025-07-08", "originalPath": "staging/172.19.0.14_2025-07-08", "path": "exercises/CA-SUL-virtual-99RC-2025-07-08/simulations/172.19.0.14_2025-07-08"}], "startDate": "2025-07-08", "trainedForce": "9", "updatedAt": "2025-07-08T16:43:43.163Z", "wikiUrl": ""}}
[2025-07-08T16:45:41.276+0000] {spark_submit.py:644} INFO - 25/07/08 16:45:41 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2025-07-08T16:45:42.909+0000] {spark_submit.py:644} INFO - Traceback (most recent call last):
[2025-07-08T16:45:42.910+0000] {spark_submit.py:644} INFO - File "/opt/bitnami/spark/jobs/process_exercises_job.py", line 26, in <module>
[2025-07-08T16:45:42.914+0000] {spark_submit.py:644} INFO - import extract_dis_data, extract_protobuf_data, extract_pdf_text
[2025-07-08T16:45:42.914+0000] {spark_submit.py:644} INFO - ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-07-08T16:45:42.915+0000] {spark_submit.py:644} INFO - ModuleNotFoundError: No module named 'extract_dis_data'
[2025-07-08T16:45:42.966+0000] {spark_submit.py:644} INFO - 25/07/08 16:45:42 INFO ShutdownHookManager: Shutdown hook called
[2025-07-08T16:45:42.967+0000] {spark_submit.py:644} INFO - 25/07/08 16:45:42 INFO ShutdownHookManager: Deleting directory /tmp/spark-e9e08a98-2706-4470-9cb3-ac2bc6406a15
[2025-07-08T16:45:43.035+0000] {taskinstance.py:3336} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 776, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 742, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 424, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 197, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 566, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://spark-master:7077 --jars /opt/bitnami/spark/jars/iceberg-spark-runtime-3.5_2.12-1.8.1.jar,/opt/bitnami/spark/jars/nessie-spark-extensions-3.5_2.12-0.103.2.jar,/opt/bitnami/spark/jars/nessie-client-0.103.2.jar,/opt/bitnami/spark/jars/hadoop-aws-3.3.2.jar,/opt/bitnami/spark/jars/aws-java-sdk-bundle-1.12.262.jar, --name arrow-spark --deploy-mode client /opt/bitnami/spark/jobs/process_exercises_job.py {"id": "CA-SUL-virtual-99RC-2025-07-08", "manifest_entry": {"data": {"createdAt": "2025-07-08T16:43:26.899Z", "origin": "CA-SUL", "path": "exercises/CA-SUL-virtual-99RC-2025-07-08"}, "id": "CA-SUL-virtual-99RC-2025-07-08"}, "metadata": {"createdAt": "2025-07-08T16:43:26.899Z", "description": "99RC", "documents": [{"description": "", "id": "1751993019656", "mimetype": "application/pdf", "name": "Op Jacui IV - Ordem de Operacoes.pdf", "path": "exercises/CA-SUL-virtual-99RC-2025-07-08/documents/Op Jacui IV - Ordem de Operacoes.pdf", "size": 336744, "uploadedAt": "2025-07-08T16:43:39.656Z"}], "endDate": "2025-07-10", "id": "CA-SUL-virtual-99RC-2025-07-08", "name": "Exerc\u00edcio Virtual", "origin": "CA-SUL", "simulationType": "virtual", "simulations": [{"id": "172.19.0.14_2025-07-08", "movedAt": "2025-07-08T16:43:43.163Z", "name": "172.19.0.14_2025-07-08", "originalPath": "staging/172.19.0.14_2025-07-08", "path": "exercises/CA-SUL-virtual-99RC-2025-07-08/simulations/172.19.0.14_2025-07-08"}], "startDate": "2025-07-08", "trainedForce": "9", "updatedAt": "2025-07-08T16:43:43.163Z", "wikiUrl": ""}}. Error code is: 1.
[2025-07-08T16:45:43.040+0000] {taskinstance.py:1242} INFO - Marking task as FAILED. dag_id=process_exercises, task_id=process_exercises, run_id=manual__2025-07-08T16:45:33.858254+00:00, execution_date=20250708T164533, start_date=20250708T164538, end_date=20250708T164543
[2025-07-08T16:45:43.160+0000] {taskinstance.py:349} INFO - ::group::Post task execution logs
[2025-07-08T16:45:43.161+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 163 for task process_exercises (Cannot execute: spark-submit --master spark://spark-master:7077 --jars /opt/bitnami/spark/jars/iceberg-spark-runtime-3.5_2.12-1.8.1.jar,/opt/bitnami/spark/jars/nessie-spark-extensions-3.5_2.12-0.103.2.jar,/opt/bitnami/spark/jars/nessie-client-0.103.2.jar,/opt/bitnami/spark/jars/hadoop-aws-3.3.2.jar,/opt/bitnami/spark/jars/aws-java-sdk-bundle-1.12.262.jar, --name arrow-spark --deploy-mode client /opt/bitnami/spark/jobs/process_exercises_job.py {"id": "CA-SUL-virtual-99RC-2025-07-08", "manifest_entry": {"data": {"createdAt": "2025-07-08T16:43:26.899Z", "origin": "CA-SUL", "path": "exercises/CA-SUL-virtual-99RC-2025-07-08"}, "id": "CA-SUL-virtual-99RC-2025-07-08"}, "metadata": {"createdAt": "2025-07-08T16:43:26.899Z", "description": "99RC", "documents": [{"description": "", "id": "1751993019656", "mimetype": "application/pdf", "name": "Op Jacui IV - Ordem de Operacoes.pdf", "path": "exercises/CA-SUL-virtual-99RC-2025-07-08/documents/Op Jacui IV - Ordem de Operacoes.pdf", "size": 336744, "uploadedAt": "2025-07-08T16:43:39.656Z"}], "endDate": "2025-07-10", "id": "CA-SUL-virtual-99RC-2025-07-08", "name": "Exerc\u00edcio Virtual", "origin": "CA-SUL", "simulationType": "virtual", "simulations": [{"id": "172.19.0.14_2025-07-08", "movedAt": "2025-07-08T16:43:43.163Z", "name": "172.19.0.14_2025-07-08", "originalPath": "staging/172.19.0.14_2025-07-08", "path": "exercises/CA-SUL-virtual-99RC-2025-07-08/simulations/172.19.0.14_2025-07-08"}], "startDate": "2025-07-08", "trainedForce": "9", "updatedAt": "2025-07-08T16:43:43.163Z", "wikiUrl": ""}}. Error code is: 1.; 7250)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3029, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 282, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3184, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3208, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 776, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 742, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 424, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 197, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 566, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://spark-master:7077 --jars /opt/bitnami/spark/jars/iceberg-spark-runtime-3.5_2.12-1.8.1.jar,/opt/bitnami/spark/jars/nessie-spark-extensions-3.5_2.12-0.103.2.jar,/opt/bitnami/spark/jars/nessie-client-0.103.2.jar,/opt/bitnami/spark/jars/hadoop-aws-3.3.2.jar,/opt/bitnami/spark/jars/aws-java-sdk-bundle-1.12.262.jar, --name arrow-spark --deploy-mode client /opt/bitnami/spark/jobs/process_exercises_job.py {"id": "CA-SUL-virtual-99RC-2025-07-08", "manifest_entry": {"data": {"createdAt": "2025-07-08T16:43:26.899Z", "origin": "CA-SUL", "path": "exercises/CA-SUL-virtual-99RC-2025-07-08"}, "id": "CA-SUL-virtual-99RC-2025-07-08"}, "metadata": {"createdAt": "2025-07-08T16:43:26.899Z", "description": "99RC", "documents": [{"description": "", "id": "1751993019656", "mimetype": "application/pdf", "name": "Op Jacui IV - Ordem de Operacoes.pdf", "path": "exercises/CA-SUL-virtual-99RC-2025-07-08/documents/Op Jacui IV - Ordem de Operacoes.pdf", "size": 336744, "uploadedAt": "2025-07-08T16:43:39.656Z"}], "endDate": "2025-07-10", "id": "CA-SUL-virtual-99RC-2025-07-08", "name": "Exerc\u00edcio Virtual", "origin": "CA-SUL", "simulationType": "virtual", "simulations": [{"id": "172.19.0.14_2025-07-08", "movedAt": "2025-07-08T16:43:43.163Z", "name": "172.19.0.14_2025-07-08", "originalPath": "staging/172.19.0.14_2025-07-08", "path": "exercises/CA-SUL-virtual-99RC-2025-07-08/simulations/172.19.0.14_2025-07-08"}], "startDate": "2025-07-08", "trainedForce": "9", "updatedAt": "2025-07-08T16:43:43.163Z", "wikiUrl": ""}}. Error code is: 1.
[2025-07-08T16:45:43.176+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2025-07-08T16:45:43.213+0000] {taskinstance.py:3924} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-07-08T16:45:43.251+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
